rust/src/dhcp/dhcp.rs

```diff
@@ -305,10 +306,9 @@ pub unsafe extern "C" fn rs_dhcp_register_parser() {
     let ip_proto_str = CString::new("udp").unwrap();
 
     if AppLayerProtoDetectConfProtoDetectionEnabled(ip_proto_str.as_ptr(), parser.name) != 0 {
-        let alproto = AppLayerRegisterProtocolDetection(&parser, 1);
-        ALPROTO_DHCP = alproto;
+        AppLayerRegisterProtocolDetection(&parser, 1);
         if AppLayerParserConfParserEnabled(ip_proto_str.as_ptr(), parser.name) != 0 {
-            let _ = AppLayerRegisterParser(&parser, alproto);
+            let _ = AppLayerRegisterParser(&parser, ALPROTO_DHCP);
```

Comment:
Likewise can now do this for all protocols that have a hardcoded value over on the C side.